cli/tests/cfncluster-unittest.py

```diff
@@ -0,0 +1,295 @@
+        self.assertEqual(test_log_stream.tell(), 0)
+
+    @mock_ec2
+    @mock_cloudformation
+    @mock_s3
+    def test_cfn_cluster_list_nonempty(self):
+        template_url = setup_configurations()
+        args = CreateClusterArgs(template_url, False)
+        cfncluster.create(args)
+        cfncluster.list(args)
+        log = test_log_stream.getvalue()
+        self.assertTrue(args.cluster_name in log)
+
+    @mock_ec2
+    @mock_cloudformation
+    @mock_autoscaling
+    @mock_s3
+    def test_cfn_cluster_update_no_reset(self):
+        template_url = setup_configurations()
+        args = UpdateClusterArgs(template_url, True, False)
+        cfncluster.create(args)
+        cfncluster.update(args)
+        success_message = 'INFO:cfncluster.cfncluster:Status: UPDATE_COMPLETE'
+        log = test_log_stream.getvalue()
+        self.assertTrue(success_message in log)
+        error_prefix = "CRITICAL:"
+        self.assertFalse(error_prefix in log)
+
+    @mock_ec2
+    @mock_cloudformation
+    @mock_autoscaling
+    @mock_s3
+    def test_cfn_cluster_update_with_reset(self):
+        template_url = setup_configurations()
+        args = UpdateClusterArgs(template_url, True, True)
+        cfncluster.create(args)
+        cfncluster.update(args)
+        success_message = 'INFO:cfncluster.cfncluster:Status: UPDATE_COMPLETE'
+        log = test_log_stream.getvalue()
+        self.assertTrue(success_message in log)
+        error_prefix = "CRITICAL:"
+        self.assertFalse(error_prefix in log)
+
+    @mock_ec2
+    @mock_cloudformation
+    @mock_autoscaling
+    @mock_s3
+    def test_cfn_cluster_update_fail(self):
+        template_url = setup_configurations()
+        args = UpdateClusterArgs(template_url, True, False)
+        with self.assertRaises(SystemExit) as sys_ex:
+            cfncluster.update(args)
+        self.assertEqual(sys_ex.exception.code, 1)
+        log = test_log_stream.getvalue()
+        error_prefix = "CRITICAL:"
+        self.assertTrue(error_prefix in log)
+
+    @mock_ec2
+    @mock_cloudformation
+    @mock_s3
+    @mock_autoscaling
+    def test_cfn_cluster_delete(self):
+        template_url = setup_configurations()
+        args = CreateClusterArgs(template_url, True)
+        cfncluster.create(args)
+        with self.assertRaises(SystemExit) as sys_ex:
+            cfncluster.delete(args)
+            self.assertEqual(sys_ex.exception.code, 0)
+        success_message = "Cluster deleted successfully"
+        log = test_log_stream.getvalue()
+        self.assertTrue(success_message in log)
+
+    @mock_ec2
+    @mock_cloudformation
+    @mock_s3
+    @mock_autoscaling
+    def test_cfn_cluster_start(self):
+        template_url = setup_configurations()
+        args = CreateClusterArgs(template_url, True)
+        cfncluster.create(args)
+        cfncluster.start(args)
+        log = test_log_stream.getvalue()
+        error_prefix = "CRITICAL:"
+        success_message = "Starting compute fleet"
```

Comment:
Also check the ASG is set to the correct values (per config)

Reply:
That would require to mock out describe_stack_resources, since we would have to get_asg-instances. It was decided that should be a done a part of larger refactoring.